scheduler/filter/expr.go

```diff
@@ -74,17 +74,24 @@ func parseExprs(key string, env []string) ([]expr, error) {
 
 func (e *expr) Match(whats ...string) bool {
 	var (
-		pattern string
-		match   bool
-		err     error
+		pattern    string
+		match      bool
+		err        error
+		startIndex int
 	)
 
-	if e.value[0] == '/' && e.value[len(e.value)-1] == '/' {
+	if e.IsSoft() {
```

Comment:
What if instead of parsing the soft affinities here at `Match`-time using `startIndex`, they were parsed directly in `parseExprs` once and for all? `parseExprs` would somehow set the soft flag for the expression.